### PR TITLE
[#46][#48][Android][UI][Integrate] As a user, I can see dropdown answer

### LINKED ISSUE #46: [Android] [UI] As a user, I can see dropdown answer
## Why

Dropdown questions are one of the most common question types to allow users to select one answer. The dropdown options could be anything, such as, a set of cities, locations, or even yes/no, since they are text-based options.

Even though this is called a dropdown, to provide the mobile experience, we will use a native picker wheel for this type of question.

## Acceptance Criteria

For dropdown questions,
- Display the answers on a native picker wheel
- Highlight the selected one
- Gray out the non-selected options

## Design

[Dropdown Question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=48%3A0&scaling=min-zoom)

### LINKED ISSUE #48: [Android] [Integrate] As a user, I can see dropdown answer
## Why

Dropdown questions are one of the most common question types to allow users to select one answer. The dropdown options could be anything, such as, a set of cities, locations, or even yes/no, since they are text-based options.

Even though this is called a dropdown, to provide the mobile experience, we will use a native picker wheel for this type of question.

## Acceptance Criteria

When a Question model has display_type as `dropdown`,

- Display a dropdown's picker wheel in the Answer section
- Apply the answers as picker wheel items
    - Use the answer's text to show as the picker wheel item's title
- By default, the first answer is initially selected

## Design

[Dropdown Question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=48%3A0&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by suho**
- Resolves #46 
- Resolves #48

## What happened 👀

- Added dropdown answer view

## Insight 📝

- I used [Pager](https://google.github.io/accompanist/pager/) to manually implement the dropdown answer based on the design.

## Proof Of Work 📹

Screenshot 2022-12-15 at 14 46 20

### CODE REVIEW

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        Divider(
+            color = Color.White,
+            thickness = 1.dp,
+            modifier = Modifier.padding(top = padding)
+        )
+        Divider(
+            color = Color.White,
+            thickness = 1.dp,
+            modifier = Modifier.padding(top = 2 * padding)
+        )
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt:83]: Showing the divider like this seems hacky in my opinion, the implementation could lead to issue on different screen resolutions 🤔 Could we detect the visible item and show the divider?

Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt:83]: @hoangnguyen92dn Do you mean putting the divider in each item and then showing it like we are doing with the font style?
```kt
style = if (isHighlight) Typography.h6 else Typography.h6.copy(fontWeight = FontWeight.Normal),
```

I tried putting the divider into each page, but the result is not good enough. There will be animation on the divider:

https://user-images.githubusercontent.com/19943832/208647445-2ea6b0aa-f6b5-4932-b497-878bc94465e8.mov


I prefer to keep the current approach as the `Divider` will expand based on the parent view. I think we can focus on the parent view if there are issues with screen resolution. What do you think?

Comment (hoangnguyen92dn) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt:83]: @suho I see, Android doesn't support this kind of view originally. 

> I prefer to keep the current approach as the Divider will expand based on the parent view. I think we can focus on the parent view if there are issues with screen resolution.

🙆‍♂️ Deal 🤝 this approach looks even better.

**Review by hoangnguyen92dn — APPROVED** [review]

**Comment by luongvo** [comment]
@suho pls rebase 👋

**Comment by suho** [comment]
@luongvo Rebased! 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt [hunk 2]
```diff
@@ -23,6 +23,12 @@ data class Question(
 ) {
     val sortedAnswers: List<Answer>
         get() = answers.sortedBy { it.displayOrder }
+
+    val answerTexts: List<String>
+        get() = sortedAnswers.map { it.text ?: "" }
+
+    val answerPlaceholders: List<String>
+        get() = sortedAnswers.map { it.inputMaskPlaceholder ?: "" }
```
Comment (hoangnguyen92dn) [inline shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt:31]: `?: ""` -> `.orEmpty()`

**Review by hoangnguyen92dn — APPROVED** [review]
Just a minor changes. LGTM 🚀

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            modifier = Modifier
+                .height(170.dp)
+                .fillMaxWidth()
+        ) { page ->
+            val choice = choices[page]
+            val isHighlight = pagerState.currentPage == page
+            Box(
+                contentAlignment = Alignment.Center,
+                modifier = Modifier
+                    .height(56.dp)
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt:50]: please use common dimension

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt (hunk 2 above)
Comment (luongvo) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt:31]: we need to use https://kotlinlang.org/api/latest/jvm/stdlib/ to have extra extensions like this, it's up to @suho imo.

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt (hunk 2 above)
Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt]: Fixed in https://github.com/suho/kmm-ic/pull/121/commits/65286cf9f85f749815db7767583f78ed841ced98

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt (hunk 3 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt]: Fixed in https://github.com/suho/kmm-ic/pull/121/commits/65286cf9f85f749815db7767583f78ed841ced98

File: shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt (hunk 2 above)
Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt]: > we need to use [kotlinlang.org/api/latest/jvm/stdlib](https://kotlinlang.org/api/latest/jvm/stdlib/) to have extra extensions like this

@luongvo Just curious, does it affect anything? It seems there is no import

Comment (luongvo) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt]: @suho we need to import `const val STDLIB = "org.jetbrains.kotlin:kotlin-stdlib:${Versions.KOTLIN}"` to commonMain 😛

Comment (suho) [reply shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt]: LOL, but I did not add anything, just write `orEmpty()`

**Review by luongvo — APPROVED** [review]

**Pull request merged**